src/main/java/com/zendesk/maxwell/producer/MaxwellKinesisProducer.java

```diff
@@ -32,82 +31,169 @@
 	private final String json;
 	private final String key;
 
-	public KinesisCallback(AbstractAsyncProducer.CallbackCompleter cc, BinlogPosition position, String key, String json) {
+	private AtomicInteger successRecords;
+	private AtomicInteger errorRecords;
+	private final KinesisProducer kinesisProducer;
+	private final String kinesisStream;
+	private int attempts = 0;
+	private final int maxAttempts;
+
+	public KinesisCallback(AbstractAsyncProducer.CallbackCompleter cc, BinlogPosition position, String key, String json, AtomicInteger successRecords, AtomicInteger errorRecords, KinesisProducer kinesisProducer, String kinesisStream, int attempts, int maxAttempts) {
 		this.cc = cc;
 		this.position = position;
 		this.key = key;
 		this.json = json;
+		this.successRecords = successRecords;
+		this.errorRecords = errorRecords;
+		this.kinesisProducer = kinesisProducer;
+		this.kinesisStream = kinesisStream;
+		this.attempts = attempts;
+		this.maxAttempts = maxAttempts;
 	}
 
 	@Override
 	public void onFailure(Throwable t) {
 		logger.error(t.getClass().getSimpleName() + " @ " + position + " -- " + key);
 		logger.error(t.getLocalizedMessage());
 
-		if(t instanceof UserRecordFailedException) {
-			Attempt last = Iterables.getLast(((UserRecordFailedException) t).getResult().getAttempts());
-			logger.error(String.format("Record failed to put - %s : %s", last.getErrorCode(), last.getErrorMessage()));
-		}
+		if(attempts>=maxAttempts) {
```

Comment:
from https://github.com/awslabs/amazon-kinesis-producer/blob/master/java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducerConfiguration.java#L1216:

>  If you do not wish to lose records and prefer to retry indefinitely, set record_ttl to a large value like INT_MAX. This has the potential to cause head-of-line blocking if network issues or throttling occur. You can respond to such situations by using the metrics reporting functions of the KPL. You may also set fail_if_throttled to true to prevent automatic retries in case of throttling.

So let's just document that config setting better in mawell and ditch the custom retry logic eh?